### PR TITLE
CORE-10330: Remove spurious Kotlin dependency from Corda APIs.

### PR DESCRIPTION
**Pull request by chrisr3**
Update our Gradle scripts now that the lingering (_and out-dated!_) Kotlin dependency has been removed from our API.

### CODE REVIEW

**Review by wzur-r3 — APPROVED** [review]
LGTM

**Review by josephzunigadaly — APPROVED** [review]

**Review by mnesbit — APPROVED** [review]
OK

File: components/ledger/ledger-common-flow-api/build.gradle [hunk 1]
```diff
@@ -8,8 +8,9 @@ description 'Corda ledger common - flow api'
 dependencies {
     compileOnly 'org.osgi:osgi.annotation'
 
-    implementation platform("net.corda:corda-api:$cordaApiVersion")
-    implementation 'net.corda:corda-ledger-common'
-    implementation 'net.corda:corda-crypto'
-    implementation project(':libs:ledger:ledger-common-data')
+    api platform("net.corda:corda-api:$cordaApiVersion")
```
Comment (blsemo) [inline components/ledger/ledger-common-flow-api/build.gradle:11]: Not that I necessarily disagree, but why did you need to change this from `implementation` to `api`?

Comment (chrisr3) [reply components/ledger/ledger-common-flow-api/build.gradle:11]: Because this bundle is providing an API, which means that everything you need to compile against this API also needs to be present on the compile classpath.

This is what `api` does. As `implementation` dependencies, they would only be included on the runtime classpath.
